python/activator/middleware_interface.py

```diff
@@ -1010,3 +1012,43 @@ def _remove_from_chain(butler: Butler, chain: str, old_collections: collections.
     for old in set(old_collections).intersection(contents):
         contents.remove(old)
     butler.registry.setCollectionChain(chain, contents, flatten=False)
+
+
+def _filter_calibs_by_date(butler: Butler,
+                           collections: typing.Any,
+                           unfiltered_calibs: collections.abc.Collection[lsst.daf.butler.DatasetRef],
+                           date: datetime.datetime
+                           ) -> collections.abc.Iterable[lsst.daf.butler.DatasetRef]:
+    """Trim a set of calib datasets to those that are valid at a particular time.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        The Butler to query for validity data.
+    collections : collection expression
+        The calibration collection(s), or chain(s) containing calibration
+        collections, to query for validity data.
+    unfiltered_calibs : collection [`lsst.daf.butler.DatasetRef`]
```

Comment:
In middleware packages we've started writes types like this in docs as
```
~collections.abc.Collection` [`lsst.daf.butler.DatasetRef`]
```
to make the container type show up as a link to the standard library docs.  You don't need to adopt that here - I don't think we've made any attempt to formalize this practice - but it seemed worth mentioning in case you find that you prefer that approach, too.